test/arrayops.jl

```diff
@@ -2651,6 +2651,14 @@ end
 # Throws ArgumentError for negative dimensions in Array
 @test_throws ArgumentError fill('a', -10)
 
+@testset "setindex" begin
+    ==ₜ(_, _) = false
+    ==ₜ(x::T, y::T) where T = x == y
+
```

Comment:
Maybe add a test that `setindex` does not mutate, even if the type stays the same.
```julia
arr = [1]
@test setindex(arr, 2, 1) ==ₜ [2]
@test arr == [1]
```
Also maybe some `@inferred`?
```

Reply:
Thanks. I added them in the latest commits.